packages/web3-eth-personal/types/tests/personal-tests.ts

```diff
@@ -0,0 +1,107 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file personal-tests.ts
+ * @author Huan Zhang <huanzhang30@gmail.com>
+ * @date 2018
+ */
+
+import {Personal} from 'web3-eth-personal';
+import {HttpProvider, BatchRequest, JsonRpcMapper, JsonRpcResponseValidator} from 'web3-providers';
+import {Network} from 'web3-net';
+
+const options = {
+    timeout: 20000,
+    headers: [
+        {
+            name: 'Access-Control-Allow-Origin', value: '*'
+        }
+    ]
+};
+
+const httpProvider = new HttpProvider('http://localhost:8545', options);
+
+const batchRequest = new BatchRequest(httpProvider, JsonRpcMapper, JsonRpcResponseValidator);
+
+const PersonalClass = new Personal(
+    httpProvider,
+)
+
+// $ExpectType boolean
+PersonalClass.setProvider(httpProvider)
+
+// $ExpectType Providers
+PersonalClass.providers;
+
+// $ExpectType AbstractProviderAdapter
+PersonalClass.givenProvider;
+
+// $ExpectType AbstractProviderAdapter
+PersonalClass.currentProvider;
+
+// $ExpectType BatchRequest
+new PersonalClass.BatchRequest();
+
+// $ExpectType void
+batchRequest.add({})
+
+// $ExpectType void
+batchRequest.execute();
```

Comment:
dont't need to test this, its already been tested in web3-providers, new personal.BatchRequest() is enough